src/main/java/duke/tasks/tasktypes/EventTask.java

```diff
@@ -0,0 +1,99 @@
+package duke.tasks.tasktypes;
+
+import duke.error.exceptions.NoStateChangeException;
+import duke.io.FileManager;
+import duke.tasks.Task;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Event item that keeps track of when an item is slated to happen.
+ */
+public class EventTask extends Task {
+    /** Keyword to mark when an event is supposed to happen */
+    public static final String COMMAND_AT = "/at";
+
+    /** Icon denoting task type. */
+    private static final String TYPE_ICON = "E";
+
+    /** Contains text following the {@link EventTask#COMMAND_AT} command. */
+    private final String EVENT;
+
+    /**
+     * Splits the text into two and initializes
+     * the {@link EventTask#EVENT} string.
+     *
+     * @param text input text to be parsed
+     */
+    public EventTask(String text) {
+        super(text.split(COMMAND_AT)[0]);
+        EVENT = text.split(COMMAND_AT)[1].trim();
+    }
+
+    /**
+     * Checks if command is equal to {@link EventTask#TYPE_ICON}
+     *
+     * @param command command string
+     * @return true if it is equal
+     */
+    public static boolean isCommand(String command) {
+        return command.trim().equals(TYPE_ICON);
+    }
+
+    /**
+     * Parses string loaded from save file into a {@link EventTask} instance.
+     *
+     * @param input input to be split by separator
+     * @return {@link EventTask} instance
+     */
+    public static EventTask parseEventTask(String input) {
+        // Parse array
+        String[] sections = input.split(Pattern.quote(FileManager.getSeparator()));
+        String subcommand = sections[3];
+        String text = sections[2];
+        String icon = sections[1];
+        EventTask bufferTask = new EventTask(text + "/at" + subcommand);
```

Comment:
Similarly, since the `/at` final is already declared above, you can consider replacing the magic string here.